src/index.ts

```diff
@@ -306,8 +307,16 @@ class Geckoboard {
     return new Dataset(schema);
   }
 
-  ping(): Promise<void> {
-    return Promise.resolve();
+  async ping(): Promise<void> {
+    const base64UserString = btoa(`${this.apiKey}:`);
+    const res = await fetch('https://api.geckoboard.com', {
+      headers: {
+        Authorization: `Basic ${base64UserString}`,
+      },
+    });
+    if (res.status !== 200) {
+      throw new Error('Ping unsuccessful');
```

Comment:
I'm not certain what exactly we should throw here - e.g. is it useful for the user to know they got back a 401?

Reply:
What response does the Iris API return in that case?

Reply:
401 - `{ error: { message: 'Your API key is invalid' }}`

Reply:
We should pass on the message then. What's the best practice for errors in JS client libraries these days, a class that extends `Error`?

Reply:
in this case I'd probably just throw an `Error` where the message is the string from the API

Reply:
Cool, so you don't think a custom error class is necessary?

Reply:
nah, in practice it'd only be switching on string content anyway, so we can let the customers do that

Reply:
The custom error class could add the HTTP status we got back from the API — whether that's useful or not is another question.